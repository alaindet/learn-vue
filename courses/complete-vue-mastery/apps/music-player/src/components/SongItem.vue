<template>
  <li
    :id="`song-id-${song.docId}`"
    :class="'some-dynamic-class'"
    class="
      flex
      justify-between
      items-center
      p-3
      pl-6
      cursor-pointer
      transition
      duration-300
      hover:bg-gray-50
    "
  >
    <div>
      <!-- <router-link :to="{ name: 'song', params: { songid: song.docId } }"> -->
      <router-link :to="'/song/'+song.docId" class="font-bold block text-gray-600">
        {{ song.modifiedName }}
      </router-link>
      <span class="song-author text-gray-500 text-sm">
        {{ song.displayName }}
      </span>
    </div>

    <div class="text-gray-600 text-lg">
      <!--
      <router-link :to="{
        name: 'song',
        params: { songid: song.docId },
        hash: '#comments'
      }">
      -->
      <router-link :to="`/song/${song.docId}#comments`">
        <span class="comments">
          <i class="fa fa-comments text-gray-600"></i>
          {{ song.commentsCount }}
        </span>
      </router-link>
    </div>
  </li>
</template>

<script>
export default {
  name: 'SongItem',
  props: {
    song: {
      type: Object,
      required: true,
    },
  },
};
</script>
