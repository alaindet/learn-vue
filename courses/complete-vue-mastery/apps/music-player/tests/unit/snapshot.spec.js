import { shallowMount, RouterLinkStub } from '@vue/test-utils';

import SongItem from '@/components/SongItem.vue';

describe('Snapshots SongItem.vue', () => {
  test('renders correctly', () => {
    const song = {
      docId: 'abcdefg',
      modifiedName: 'song name',
      displayName: 'user name',
      commentsCount: 4,
    };

    const component = shallowMount(SongItem, {
      propsData: {
        song,
      },
      global: {
        components: {
          'router-link': RouterLinkStub,
        },
      },
    });

    expect(component.element).toMatchSnapshot();
  });
});
