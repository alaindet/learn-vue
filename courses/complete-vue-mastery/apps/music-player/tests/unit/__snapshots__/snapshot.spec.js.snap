// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots SongItem.vue renders correctly 1`] = `
<li
  class="some-dynamic-class flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  id="song-id-abcdefg"
>
  <div>
    <!-- &lt;router-link :to="{ name: 'song', params: { songid: song.docId } }"&gt; -->
    <router-link-stub
      class="font-bold block text-gray-600"
      to="/song/abcdefg"
    />
    <span
      class="song-author text-gray-500 text-sm"
    >
      user name
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <!--
      &lt;router-link :to="{
        name: 'song',
        params: { songid: song.docId },
        hash: '#comments'
      }"&gt;
      -->
    <router-link-stub
      to="/song/abcdefg#comments"
    />
  </div>
</li>
`;
